### DISCUSSION TITLE
Creating an array for the dependent variable

### DISCUSSION THREAD

**Discussion by jasperott** [stocnet/rsiena#29] [Q&A] [answered]
Hi All, 

For my thesis I'm working with RSiena. I am currently developing an inventor-patent network (two inventors are linked when they co-developed a patent). The network is growing and in the first wave I have a matrix of [1:38, 1:38], the second wave [1:68, 1:68] and the third wave has [1:110, 1:110] actors. My question now is, which dimensions do I need to use when creating the array for the dependent variable. 

Right now, I use the following code: 
```
ThesisData <- array( c( MA04, MA05, MA06 ),
                     dim = c( 110, 110, 3 ) )
```

Is this the right way to go? I am wondering whether someone of you could help me out?

Thank you very much in advance!

**Comment by jhollway** [comment, marked as answer]
Yes, the array constructed should be maximal, i.e. including a row (and, if one-mode, column) for each actor that appears in any wave. Then you are recommended to use either composition change or structural zeros to constrain the simulations such that only actors that you know are in the network at a particular wave are offered opportunities to modify their local network. There are more details about how these are constructed in [the manual](http://www.stats.ox.ac.uk/~snijders/siena/).

**Comment by jasperott** [comment]
Thanks a lot, this will help!
